src/api/wayfire/signal-definitions.hpp

```diff
@@ -75,6 +75,22 @@ struct view_self_request_focus_signal : public _view_signal
 {
 };
 
+/**
+ * view-hints-changed signal is emitted on the view's output whenever the client indicates the
+ * views hints have changed (example urgency hint).
```

Comment:
What is the difference between hints-changed and system-bell, except that they come from different sources?

Reply:
well, if you have gnome terminal and test it on gnome, you will see that if your cursor is already most-left at the terminal and press left you will hear woof if that is your alert sound.
hints changed is really like IM message received

Reply:
Maybe a stupid question, but is this different from self-focus-request? My understanding was that the two are very similar.

Reply:
Yes it's different :) 
The hint is not necessarily a request for focus.
It can be interpreted by a notify-send, a blinking taskbar icon, a sound,
whatever.

Reply:
Also I kept it as view-hints-changed not view-urgency or something as there maybe other kind of hints